clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -564,9 +585,9 @@ struct BuiltinTypeMethodBuilder {
                                       OK_Ordinary);
   }
 
-  BuiltinTypeMethodBuilder &
-  callBuiltin(StringRef BuiltinName, ArrayRef<Expr *> CallParms,
-              bool AddResourceHandleAsFirstArg = true) {
+  template <typename... Ts>
+  BuiltinTypeMethodBuilder &callBuiltin(StringRef BuiltinName, Ts... ArgSpecs) {
+    SmallVector<Expr *> Args{convertPlaceholder(std::forward<Ts>(ArgSpecs))...};
```

Comment:
Is SmallVector smart enough for this to evaporate into being a stack allocated array, or is there a chance that it might make a heap allocation to store the args?

Reply:
Yes, `SmallVector` will have stack allocated storage.  Its size is derived from the initialization, which is dependent on the number of ArgSpecs.

Reply:
This raises a good point though - I'll change this to just use `std::array` here since we know the exact size at compile time.